src/main/java/sg/ncl/MainController.java

```diff
@@ -2686,13 +2705,21 @@ public String updateExperiment(@PathVariable String teamId, @PathVariable String
     }
 
     @PostMapping("/update_experiment/{teamId}/{expId}")
-    public String updateExperimentFormSubmit(@ModelAttribute("edit_experiment") Experiment2 editExperiment, @PathVariable String teamId, @PathVariable String expId, RedirectAttributes redirectAttributes) throws WebServiceRuntimeException {
+    public String updateExperimentFormSubmit(@ModelAttribute("edit_experiment") Experiment2 editExperiment, BindingResult bindingResult, @PathVariable String teamId, @PathVariable String expId, RedirectAttributes redirectAttributes) throws WebServiceRuntimeException {
+
+        // check max duration for errors
+        if (bindingResult.hasErrors() || !editExperiment.getMaxDuration().toString().matches("\\d+")) {
+            redirectAttributes.addFlashAttribute(MESSAGE, MAX_DURATION_ERROR);
+            return "redirect:/update_experiment/" + teamId + "/" + expId;
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "redirect:/update_experiment/" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)